site/static/examples/ingress/usage.yaml

```diff
@@ -51,7 +51,7 @@ kind: Ingress
 metadata:
   name: example-ingress
   annotations:
-    ingress.kubernetes.io/rewrite-target: /
+    nginx.ingress.kubernetes.io/rewrite-target: /
```

Comment:
What versions is this compatible with?

Reply:
we should also make sure we don't break contour.
although looking at https://github.com/projectcontour/contour/issues/64 i don't think they ever used that annotation.

basic experimentation with removing that annotation altogether also worked for me both on contour and nginx.
we should probably just drop it? to have a minimal example as much as possible.

Reply:
I've tried this with kubernetes `1.17`. If you are looking for ingress controller version, it is `0.30.0`. Any specific version you want me to try this out ? 

I see the issue here. Let me try out contour as well. Probably we need to break this ingress spec into two parts. Nginx specific and Contour specific. 
```
---
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: example-ingress
  annotations:
    ingress.kubernetes.io/rewrite-target: /
spec:
  rules:
  - http:
      paths:
      - path: /foo
        backend:
          serviceName: foo-service
          servicePort: 5678
      - path: /bar
        backend:
          serviceName: bar-service
          servicePort: 5678
```

This documentation might seem trivial and easy to be resolved for someone who knows this. But I couldn't get pass it and had to ask in slack. So I feel this should be clear enough for anyone trying this out?

Reply:
I'm sorry to bug you both for such a simple issue. But I can't get ingress working with removing this annotation. I'm getting 404 ( for both contour and nginx-ingress).  Below is how I'm creating my cluster 
```
cat <<EOF | kind create cluster --config -
kind: Cluster
apiVersion: kind.x-k8s.io/v1alpha4
nodes:
- role: control-plane
  kubeadmConfigPatches:
  - |
    kind: InitConfiguration
    nodeRegistration:
      kubeletExtraArgs:
        node-labels: "ingress-ready=true"
        authorization-mode: "AlwaysAllow"
  extraPortMappings:
  - containerPort: 80
    hostPort: 80
    protocol: TCP
  - containerPort: 443
    hostPort: 443
    protocol: TCP
EOF
```

Reply:
the answer seems to be 0.22 ?
https://kubernetes.github.io/ingress-nginx/examples/rewrite/#rewrite-target

Reply:
Yes. For ingress after 0.22 we must add ingress annotation `nginx.ingress.kubernetes.io/rewrite-target`. Hence following current documentation fails. 

Reason for this is our documentation mentions `ingress.kubernetes.io/rewrite-target` (notice missing **nginx**). Fixing this at common ingress manifest can be an issue with `Contour` example. I tried my hand with contour but couldn't get it working using our doc.